src/hotspot/share/opto/regmask.hpp

```diff
@@ -158,9 +161,13 @@ class RegMask {
   // The last bit in the register mask indicates that the mask should repeat
   // indefinitely with ONE bits.  Returns TRUE if mask is infinite or
   // unbounded in size.  Returns FALSE if mask is finite size.
-  bool is_AllStack() const { return _RM_UP[_RM_SIZE - 1U] >> (_WordBits - 1U); }
+  bool is_AllStack() const {
+    return (_RM_UP[_RM_SIZE - 1U] & (uintptr_t(1) << (_WordBits - 1U))) != 0;
+  }
 
-  void set_AllStack() { Insert(OptoReg::Name(CHUNK_SIZE-1)); }
+  void set_AllStack() {
+    _RM_UP[_RM_SIZE - 1U] |= (uintptr_t(1) << (_WordBits - 1U));
+  }
```

Comment:
`_WordBits - 1U` is used in several places in this file. Should we add and use new _WordBitsM1 value instead?